dev-libs/aws-sdk-cpp/aws-sdk-cpp-1.6.23.ebuild

```diff
@@ -13,26 +13,32 @@ LICENSE="Apache-2.0"
-
-for module in ${ALL_AWS_TARGETS}; do
-	if [ "$module" = "core" ]; then
-		IUSE_AWS_TARGETS+=" aws_targets_${module}"
-	else
-		IUSE_AWS_TARGETS+=" -aws_targets_${module}"
-	fi
+MODULES="aws_access-management aws_acm aws_acm-pca aws_alexaforbusiness aws_apigateway aws_application-autoscaling aws_appstream aws_appsync aws_athena aws_autoscaling aws_autoscaling-plans aws_AWSMigrationHub aws_batch aws_budgets aws_ce aws_cloud9 aws_clouddirectory aws_cloudformation aws_cloudfront aws_cloudhsm aws_cloudhsmv2 aws_cloudsearch aws_cloudsearchdomain aws_cloudtrail aws_codebuild aws_codecommit aws_codedeploy aws_codepipeline aws_codestar aws_cognito-identity aws_cognito-idp aws_cognito-sync aws_comprehend aws_config aws_connect aws_cur aws_datapipeline aws_dax aws_devicefarm aws_directconnect aws_discovery aws_dms aws_ds aws_dynamodb aws_dynamodbstreams aws_ec2 aws_ecr aws_ecs aws_eks aws_elasticache aws_elasticbeanstalk aws_elasticfilesystem aws_elasticloadbalancing aws_elasticloadbalancingv2 aws_elasticmapreduce aws_elastictranscoder aws_email aws_es aws_events aws_firehose aws_fms aws_gamelift aws_glacier aws_glue aws_greengrass aws_guardduty aws_health aws_iam aws_identity-management aws_importexport aws_inspector aws_iot aws_iot1click-devices aws_iot1click-projects aws_iotanalytics aws_iot-data aws_iot-jobs-data aws_kinesis aws_kinesisanalytics aws_kinesisvideo aws_kinesis-video-archived-media aws_kinesis-video-media aws_kms aws_lambda aws_lex aws_lex-models aws_lightsail aws_logs aws_machinelearning aws_macie aws_marketplacecommerceanalytics aws_marketplace-entitlement aws_mediaconvert aws_medialive aws_mediapackage aws_mediastore aws_mediastore-data aws_mediatailor aws_meteringmarketplace aws_mobile aws_mobileanalytics aws_monitoring aws_mq aws_mturk-requester aws_neptune aws_opsworks aws_opsworkscm aws_organizations aws_pi aws_pinpoint aws_polly aws_pricing aws_queues aws_rds aws_redshift aws_rekognition aws_resource-groups aws_resourcegroupstaggingapi aws_route53 aws_route53domains aws_s3 aws_s3-encryption aws_sagemaker aws_sagemaker-runtime aws_sdb aws_secretsmanager aws_serverlessrepo aws_servicecatalog aws_servicediscovery aws_shield aws_sms aws_snowball aws_sns aws_sqs aws_ssm aws_states aws_storagegateway aws_sts aws_support aws_swf aws_text-to-speech aws_transcribe aws_transfer aws_translate aws_waf aws_waf-regional aws_workdocs aws_workmail aws_workspaces aws_xray"
```

Comment:
Secondly, I really do think you should kill all those flags. Having that many flags is just insane, and no reasonable Gentoo user will spend his time going through them all.

Reply:
What would you recommend as an alternate? Most users will need a subset of these modules and not all of them, so I'd like the package to retain that choice.

Second, all modules must be built with identical use flags (specifically, static libs and memory management), so I'm not sure how to restrict that if I split this package across multiple smaller ones.

Reply:
What is the cost of building all modules unconditionally?

Reply:
None apart from compile time and space used. core is required by all, so for example aws polly + core takes around 30 seconds, but the entire package takes more like 30 minutes.

I am not sure on space requirements but I think given current hardware, it is comparatively insignificant.

Reply:
I wish. AWS does however offer as many individual services. Of course some of these services work in conjunction with others, but there are no harddeps afaik

Reply:
Ok, so let's stay with the flags for now. Please wrap this for a start, do all the other stuff and I'll take a second look.